src/test/scala/org/broadinstitute/dsde/workbench/sam/service/TosServiceSpec.scala

```diff
@@ -29,82 +33,137 @@ class TosServiceSpec extends AnyFlatSpec with TestSupport with BeforeAndAfterAll
+    assertResult(expected = false, s"getTosStatus(${defaultUser.id}) should have returned false")(actual = adherenceStatus.userHasAcceptedLatestTos)
   }
 
-  "TosService.isTermsOfServiceStatusAcceptable" should "allow all requests to the API if TOS is disabled" in {
+  "TosService" should "allow users to accept new version of ToS" in {
     assume(databaseEnabled, databaseEnabledClue)
+
     dirDAO.createUser(defaultUser, samRequestContext).unsafeRunSync()
-    val tosServiceDisabled = new TosService(dirDAO, "example.com", TestSupport.tosConfig.copy(enabled = false))
-    tosServiceDisabled.isTermsOfServiceStatusAcceptable(defaultUser) should be(true)
+
+    // Accept V0
+    val tosServiceV0 = new TosService(dirDAO, TestSupport.tosConfig.copy(version = "0"))
+    tosServiceV0.getTosAdherenceStatus(defaultUser).unsafeRunSync() shouldBe TermsOfServiceAdherenceStatus(defaultUser.id, false, false)
+    tosServiceV0.acceptTosStatus(defaultUser.id, samRequestContext).unsafeRunSync() shouldBe true
+    val userAcceptedV0 = dirDAO.loadUser(defaultUser.id, samRequestContext).unsafeRunSync().orNull
+    val adherenceStatusV0 = tosServiceV0.getTosAdherenceStatus(userAcceptedV0).unsafeRunSync()
+    adherenceStatusV0.userHasAcceptedLatestTos shouldBe true
+
+    // Accept V2
+    val tosServiceV2 = new TosService(dirDAO, TestSupport.tosConfig.copy(version = "2"))
+    val adherenceStatusNotAcceptedLatest = tosServiceV2.getTosAdherenceStatus(userAcceptedV0).unsafeRunSync()
+    adherenceStatusNotAcceptedLatest.userHasAcceptedLatestTos shouldBe false
+    adherenceStatusNotAcceptedLatest.acceptedTosAllowsUsage shouldBe false
+
+    tosServiceV2.acceptTosStatus(userAcceptedV0.id, samRequestContext).unsafeRunSync() shouldBe true
+    val userAcceptedV2 = dirDAO.loadUser(userAcceptedV0.id, samRequestContext).unsafeRunSync().orNull
+    val adherenceStatusV2 = tosServiceV2.getTosAdherenceStatus(userAcceptedV2).unsafeRunSync()
+    adherenceStatusV2.userHasAcceptedLatestTos shouldBe true
+    adherenceStatusV2.acceptedTosAllowsUsage shouldBe true
   }
 
-  it should "not allow users who have never accepted a ToS version to use the API, even with a grace period enabled" in {
+  it should "always allow service account users to use Terra" in {
     assume(databaseEnabled, databaseEnabledClue)
-    dirDAO.createUser(defaultUser, samRequestContext).unsafeRunSync()
     val tosServiceGracePeriodEnabled =
-      new TosService(dirDAO, "example.com", TestSupport.tosConfig.copy(enabled = true, version = "2", isGracePeriodEnabled = true))
-    tosServiceGracePeriodEnabled.isTermsOfServiceStatusAcceptable(defaultUser) should be(false)
+      new TosService(mockDirDAO, TestSupport.tosConfig.copy(version = "2"))
+    val adherenceStatus = tosServiceGracePeriodEnabled.getTosAdherenceStatus(serviceAccountUser).unsafeRunSync()
+    adherenceStatus.userHasAcceptedLatestTos shouldBe false
```

Comment:
I don't think we need this assertion.

Reply:
Since we're testing the results matrix, I think its good to make sure all outputs are as expected